python/lsst/ts/m2com/controller_cell.py

```diff
@@ -0,0 +1,475 @@
+            Period to check the connection status in second. (the default is 1)
+        **kwargs : `dict`, optional
+            Additional keyword arguments in "process_lost_connection" function
+            call.
+        """
+        self._task_connection_monitor_loop = asyncio.create_task(
+            self._connection_monitor_loop(
+                process_lost_connection, *args, period=period, **kwargs
+            )
+        )
+
+    async def _connection_monitor_loop(
+        self, process_lost_connection, *args, period=1, **kwargs
+    ):
+        """Actively monitor the connection status from component. Disconnect
+        the system if the connection is lost by itself.
+
+        Parameters
+        ----------
+        process_lost_connection : `func` or `coroutine`
+            Function to process the lost of connection.
+        *args : `args`
+            Arguments needed in "process_lost_connection" function call.
+        period : `float` or `int`, optional
+            Period to check the connection status in second. (the default is 1)
+        **kwargs : `dict`, optional
+            Additional keyword arguments in "process_lost_connection" function
+            call.
+        """
+
+        self.log.debug("Begin to run the connection monitor loop from component.")
+
+        is_coroutine_function = is_coroutine(process_lost_connection)
+
+        were_clients_connected = False
+        while self.run_loops:
+
+            if self.are_clients_connected():
+                were_clients_connected = True
+            else:
+                if were_clients_connected:
+                    were_clients_connected = False
+                    self.log.info(
+                        "Lost the TCP/IP connection in the active monitoring loop."
+                    )
+
+                    await self.close()
+
+                    # Process the lost of connection
+                    if is_coroutine_function:
+                        await process_lost_connection(*args, **kwargs)
+                    else:
+                        process_lost_connection(*args, **kwargs)
+
+            await asyncio.sleep(period)
+
+        self.log.debug("Connection monitor loop from component closed.")
+
+    async def close_tasks(self):
+        """Close the asynchronous tasks.
+
+        Note: this function is safe to call even though there is no connection.
+        """
+
+        await self._close_controller_and_mock_server()
+
+        try:
+            await self.stop_loops()
+        except Exception:
+            self.log.exception("Exception while stopping the loops. Ignoring...")
+
+    async def _close_controller_and_mock_server(self):
+        """Close the controller and mock server."""
+
+        await self.close()
+
+        if self.mock_server is not None:
+            # Wait some time to let the mock server notices the controller has
+            # closed the connection.
+            await asyncio.sleep(10)
```

Comment:
Consider turning the 10 here into a constant or class attribute.

Reply:
Fixed in the commit: [449e880](https://github.com/lsst-ts/ts_m2com/pull/10/commits/449e8806e72c01fd253fbbdfc896198514aad657). Thanks!